streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```
@@ -0,0 +1,122 @@
+        memtableMinFlushTimeSensor = RocksDBMetrics.memtableMinFlushTimeSensor(streamsMetrics, metricContext);
+        memtableMaxFlushTimeSensor = RocksDBMetrics.memtableMaxFlushTimeSensor(streamsMetrics, metricContext);
+        writeStallDurationSensor = RocksDBMetrics.writeStallDurationSensor(streamsMetrics, metricContext);
+        blockCacheDataHitRatioSensor = RocksDBMetrics.blockCacheDataHitRatioSensor(streamsMetrics, metricContext);
+        blockCacheIndexHitRatioSensor = RocksDBMetrics.blockCacheIndexHitRatioSensor(streamsMetrics, metricContext);
+        blockCacheFilterHitRatioSensor = RocksDBMetrics.blockCacheFilterHitRatioSensor(streamsMetrics, metricContext);
+        bytesReadDuringCompactionSensor = RocksDBMetrics.bytesReadDuringCompactionSensor(streamsMetrics, metricContext);
+        bytesWrittenDuringCompactionSensor =
+            RocksDBMetrics.bytesWrittenDuringCompactionSensor(streamsMetrics, metricContext);
+        compactionTimeAvgSensor = RocksDBMetrics.compactionTimeAvgSensor(streamsMetrics, metricContext);
+        compactionTimeMinSensor = RocksDBMetrics.compactionTimeMinSensor(streamsMetrics, metricContext);
+        compactionTimeMaxSensor = RocksDBMetrics.compactionTimeMaxSensor(streamsMetrics, metricContext);
+        numberOfOpenFilesSensor = RocksDBMetrics.numberOfOpenFilesSensor(streamsMetrics, metricContext);
+        numberOfFileErrorsSensor = RocksDBMetrics.numberOfFileErrorsSensor(streamsMetrics, metricContext);
+    }
+
+    public void addStatistics(final String statisticsId,
+                              final Statistics statistics,
+                              final StreamsMetricsImpl streamsMetrics,
+                              final TaskId taskId) {
+        if (statisticsToRecord.isEmpty()) {
+            init(streamsMetrics, taskId);
+        }
+        final Statistics previousStatistics = statisticsToRecord.get(statisticsId);
```

Would we ever allow to override-register a statistics? That would only happen if we ever has two stores with the same store name right? Should that be a bug and be thrown as an exception?

---

Let's throw an exception. I guess I overthought this a bit.